source/sonata_simulation.rst

```diff
@@ -267,6 +267,7 @@ Spike events are created from the cells indicated in a file and delivered to the
    ============================== ========== ============ ==========================================
    spike_file                     text       Mandatory    Indicates the location of the file with the spike info for injection.
    source                         text       Optional     The node set to replay spikes from.
+   connection_type                text       Optional     Type of the connectivity between the source and target of injection. Supported values are :ref:`the connection types of populations<edge_pop_type>`: "chemical", "electrical", "synapse_astrocyte", "endfoot", "neuromodulatory". Default is "chemical".
```

Comment:
why do we need "connection_type" ? is that used as a filter ?
if that is the case, I would imagine a population of edges here

Reply:
Yes, it is used as a filter, same as the BlueConfig key ["Type"](https://sonata-extension.readthedocs.io/en/latest/blueconfig.html#blueconfigsection-3). For Blueconfig, it happened with the old circuit format that there could be multiple connectivity types for the same `src_population -> dst_population` so that a "Type" key is needed to select the right connectivity for the stimulus inject (by far, neurodamus only use that for replay). For Sonata circuit, I am thinking to keep this optional key for replay in case of such special scenario. If, SONATA allows only one connectivity type between populations, we won't need this key I guess.

Reply:
The challenge is that things are a bit more complicated with what sonata enables, let me give you an example:
We have mid-range and local connectivity in the SSCx: 2 different edge population.
Both source and target populations are the same for the 2 connectivity
Both of them are "chemical" connections type.

Therefore, "connection_type" can not discriminate between the 2 populations of edges.

That is why a edge population filter seems to give finer control.

I don't know if someone is using that filter. Let's ping @MWolfR  knows about it and can tell what is appropriate.

Reply:
Hmm...sorry to ask again:
I've looked up in the SONATA spec, is it the naming convention for the `edge population`?
`/edges/${source_population}__${target_population}__${connection_type}` 
If so, how can we have different edge populations for mid-range and local connectivity in the SSCx with the same source and target populations, as well as the connection type? Or is there going to be a different definition of `edge population` for that?

Reply:
This is a fair question. This is a naming convention (just to help the user ) but this is not how the information is stored.
The ground truth is the following:
- The source and target population name are stored as a h5 attribute.
- The connection type is defined in the circuit config.

In the case of midrange connection, the population name is "midrange__S1nonbarrel_neurons__chemical", which actually does not follow the convention - as the names would collide -

Reply:
Hello @jdcourcol , 
I discussed with @ferdonline, and we think that actually such a filter for replay is not needed for SONATA circuits:
1) For sonata circuits, there should be one unique connectivity type between the `src_population -> dst_population` path, so that for replay there is no confusion about which connectivity to inject. (This is different from the old-format circuits)
2) For the scenario of mid-range and local connectivity that you pointed out above (same src population, same dst population, same connectivity type), neurodamus internally treats edges from the two ranges as the same connection type. For replay, there is no need to have an additional filter. Just with the already defined parameters `source` and `node_set` for `synapse_replay` input, neuordamus can find the correct connection between the two ranges for the replay. 

If you agree, I will remove this change from this PR.
cc @jamesgking

Reply:
> For sonata circuits, there should be one unique connectivity type between the src_population -> dst_population path, so that for replay there is no confusion about which connectivity to inject. (This is different from the old-format circuits)

I am not sure about that: You can have src_pop -> dst_pop, but some one edge population is chemical and the other one is gap_junction.

Reply:
We should probably remove that for the time being as you suggest, as long as the use case / user intent is not clearly defined.
Same when we migrate from blueconfig to sonata.
@jamesgking ?

Reply:
@JDC, @jamesgking  Ok, I will not add the new key `connection_type` for replay in this PR. We will discuss later when we have a clearly defined use case. The PR has been updated.